fastlane/actions/project_name.rb

```diff
@@ -0,0 +1,71 @@
+require 'xcodeproj'
+
+module Fastlane
+  module Actions
+    class ProjectNameAction < Action
+
+      PROJECT_EXTENSION = ".xcodeproj".freeze
+
+      # This script returns the application name
+      # if there is a scheme that matches the project name.
+      # Otherwise, it assumes the project name should be specified 
+      # by the user and fails.
+
+      def self.run(params)
+        default_project_name
+      end
+
+      # Fastlane Action class required functions.
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :project_name, optional: true, default_value: default_project_name)
+        ]
+      end
+
+      # Available options default_value helpers
+
+      # In case there is a single '.xcodeproj' in the default directory
+      # it can be automatically inferred by the script 
+      # if no parameter project is received.
+      def self.default_project
+        projects = Dir.entries('.').select { |each| File.extname(each) == PROJECT_EXTENSION }
```

Comment:
Just to check: Does extname return the `.` as well? More explicitly, would it return `xcodeproj` or `.xcodeproj`. (Because we are comparing to `.xcodeproj.freeze`, which I guess is with the dot right? If it's notwith the dot then we should review `self.default_project_filename` that just concatenates withput adding a dot).

Reply:
Yes, it returns the extension including the `.`